Integrations/MicrosoftGraphGroups/MicrosoftGraphGroups.py

```diff
@@ -0,0 +1,577 @@
+        """Get the Microsoft Graph Access token from the instance token or generates a new one if needed.
+
+        Returns:
+            The access token.
+        """
+        integration_context = demisto.getIntegrationContext()
+        access_token = integration_context.get('access_token')
+        valid_until = integration_context.get('valid_until')
+        if access_token and valid_until:
+            if epoch_seconds() < valid_until:
+                return access_token
+
+        dbot_response = requests.post(
+            self.token_retrieval_url,
+            headers={'Accept': 'application/json'},
+            data=json.dumps({
+                'app_name': APP_NAME,
+                'registration_id': self.auth_id,
+                'encrypted_token': get_encrypted(self.tenant, self.enc_key)
+            }),
+            verify=self.use_ssl
+        )
+        if dbot_response.status_code not in {200, 201}:
+            msg = 'Error in authentication. Try checking the credentials you entered.'
+            try:
+                demisto.info(f'Authentication failure from server: {dbot_response.status_code}'
+                             f' {dbot_response.reason} {dbot_response.text}')
+                err_response = dbot_response.json()
+                server_msg = err_response.get('message')
+                if not server_msg:
+                    title = err_response.get('title')
+                    detail = err_response.get('detail')
+                    if title:
+                        server_msg = f'{title}. {detail}'
+                if server_msg:
+                    msg += f' Server message: {server_msg}'
+            except Exception as err:
+                demisto.error(f'Failed parsing error response - Exception: {err}')
+            raise Exception(msg)
+        try:
+            gcloud_function_exec_id = dbot_response.headers.get('Function-Execution-Id')
+            demisto.info(f'Google Cloud Function Execution ID: {gcloud_function_exec_id}')
+            parsed_response = dbot_response.json()
+        except ValueError:
+            raise Exception(
+                'There was a problem in retrieving an updated access token.\n'
+                'The response from the Demistobot server did not contain the expected content.'
+            )
+        access_token = parsed_response.get('access_token')
+        expires_in = parsed_response.get('expires_in', 3595)
+        time_buffer = 5  # seconds by which to shorten the validity period
+        if expires_in - time_buffer > 0:
+            # err on the side of caution with a slightly shorter access token validity period
+            expires_in = expires_in - time_buffer
+
+        demisto.setIntegrationContext({
+            'access_token': access_token,
+            'valid_until': epoch_seconds() + expires_in
+        })
+        return access_token
+
+    def http_request(self, method: str, url_suffix: str = None, params: Dict = None, body: Optional[str] = None,
```

Comment:
why dont we use the generic `http_request` ? just because of the token and the url?

Reply:
there is also a next_link param in MS Graph, see arseny's doc in the confluence.